<a href="https://colab.research.google.com/github/venkateshblks/Hypothesis-Testing/blob/main/Hypothesis_Testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#### <b>1 .</b> A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [ ]:
import pandas as pd
cutlets=pd.read_csv('Cutlets.csv')
cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

**To analyze the data and draw inferences, we can use a two-sample t-test**

**Null Hypothesis (H0):  μ1=μ2**

**There is no significant difference in the diameter of the cutlets between the two units.**
<br>
<br>
**Alternative Hypothesis (H1): μ1 is not equal to μ2**  

**There is a significant difference in the diameter of the cutlets between the two units.**

In [ ]:
from scipy import stats
alpha=0.05 #sigficance level
stats.ttest_ind(cutlets['Unit A'],cutlets['Unit B'])



TtestResult(statistic=0.7228688704678063, pvalue=0.4722394724599501, df=68.0)

**Here p_value is greater than alpha so we fail to reject null hypothesis**

**There is no significant difference in the diameter of the cutlets between the two units.**

7.0190914285714285

#### <b>2 .</b> A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.

#### Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

**we can perform an Analysis of Variance (ANOVA) test. The ANOVA test is used to compare the means of more than two groups**

**H0:μ1=μ2=μ3=μ4**

**Ha:At least one μi is different  μ**

In [14]:
tat=pd.read_csv('LabTAT.csv')
tat.head()


Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [8]:
stats.f_oneway(tat['Laboratory 1'],tat['Laboratory 2'],tat['Laboratory 3'],tat['Laboratory 4'])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

**p-value (2.11 * 10^-57 which is almost equal to zero) less than the chosen significance level (0.05), we reject the null hypothesis and conclude that there is a significant difference in average TAT among the laboratories**

#### <b>4 . </b>Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [25]:
br=pd.read_csv('BuyerRatio.csv')
br

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

**H0 : The male-female buyer rations are similar across regions**


**H1 : The male-female buyer rations are not similar across regions**

In [26]:
br.set_index('Observed Values',inplace=True)
br.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [27]:
from scipy.stats import chi2_contingency
chi2_contingency(br.values)

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

**P value of Chi-Square test is 0.66 > alpha**

**Fail to reject null hypothesis**

**The male-female buyer rations are similar across regions**

#### <b>4 . </b>TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [29]:
cof=pd.read_csv('Costomer+OrderForm.csv')
cof.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

**H0 : The defective % does not varies by centre**

**H1 : The defective % does varies by centre**

In [38]:
cof.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [41]:
dd=cof.melt(var_name='Center', value_name='Condition')
dd

Center   Condition
0     Phillippines  Error Free
1     Phillippines  Error Free
2     Phillippines  Error Free
3     Phillippines  Error Free
4     Phillippines  Error Free
...            ...         ...
1195         India  Error Free
1196         India  Error Free
1197         India  Error Free
1198         India  Error Free
1199         India  Error Free

[1200 rows x 2 columns]

In [40]:
contingency_table=pd.crosstab(
    dd['Center'],
    dd['Condition']
)

In [42]:
alpha=0.05
stats.chi2_contingency(contingency_table)

Chi2ContingencyResult(statistic=3.8589606858203545, pvalue=0.2771020991233144, dof=3, expected_freq=array([[ 28.25, 271.75],
       [ 28.25, 271.75],
       [ 28.25, 271.75],
       [ 28.25, 271.75]]))

**P value of Chi-Square test is 0.277 > alpha**

**Fail to reject null hypothesis**

**There is no association between the center and the defective status of order forms**